intents/time_compare.py

```diff
@@ -0,0 +1,180 @@
+            summary operator is applied on metric. Metric could a column
+            containing strings, if we are applying count operator on it.
+        dimensions: Type-list of str
+            It is the name of column we want.
+            'compare batsman A and B according to total_runs',
+             dimension is 'batsman'. we group by dimensions.
+        all_dimension: Type-list of str
+            It is the list of dimension columns in the initial table
+        date_range: Type-tuple
+            Tuple of start_date and end_date
+        date_column_name: Type-str
+            It is the name of column which contains date
+        date_format: Type-str
+            It is required by datetime.strp_time to parse the date in the format
+            Format Codes
+https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior
+        slices: Type-List of tuples
+            Tuple represents the conditon to keep the row.
+            (column_name, filter, value)
+            column_name - is the value of the column that the
+            condition is applied upon.
+            filter - Filters enum members, ex. Filters.IN
+        time_compare_column: Type-list of different data types
+            first element denotes the column name by which we will do comparision.
+            second and third element are a tuple of start and end date range.
+            last element is date_formate in which the date of columns present.
+        summary_operator: Type-summary_operators enum members
+            It denotes the summary operator, after grouping by dimensions.
+            ex. SummaryOperators.MAX, SummaryOperators.SUM
+
+    Note-summary_operator is always applied on metric column passed,
+         and only when grouping is done
+    Returns:
+        The function will return both suggestions and the results in a tuple.
+        (results, suggestions)
+        results: Type - pandas dataframe, The results of the intended slice-compare
+        suggestions: Type - List of strings, List of suggestions.
+    """
+    date_column_name = kwargs.get('date_column_name', 'date')
+    date_range = kwargs.get('date_range', None)
+    date_format = kwargs.get('date_format', '%Y-%m-%d')
+
+    slices = kwargs.get('slices', None)
+
+    result_table = _time_compare_results(table, metric, dimensions, 
+                                         time_compare_column, summary_operator, 
+                                         date_column_name = date_column_name,
+                                         date_range = date_range,
+                                         date_format = date_format,
+                                         slices = slices)
+    return result_table
+
+def _time_compare_results(table, metric, dimensions, time_compare_column,
+                                             summary_operator, **kwargs):
+
+    """ This function returns both the results according to the intent
+    as well as the debiasing suggestions.
```

Comment:
The function's description is not consistent with it's functionality. It does/will not return suggestions as it;s name suggest time_compare\_**results**.
You should change it to
"
 This function returns the results according to the intent....
"